### PR TITLE
Lint out line height usage and refactor uses

### PR DESCRIPTION
**Pull request by gtrufitt**
## What does this change?
Adds `line-height` to the CSS property blacklist. Line heights should mostly come from pasteup typography rules. 

In most cases where line height has been used it wasn't being used by the browser for various reasons (overridden in cascade, same as initial).

Sometimes they're used for layout - in these cases here, I've marked as a 'Design System' so that it can have more holistic thinking around the layout primitives required.

## Why?
Consistency in design, simpler to maintain and less cognitive load on the developer.

## Link to supporting Trello card
https://trello.com/c/EcVAyY2j/527-enforcing-pasteup-%E2%97%8F%E2%97%8F%E2%97%8F%E2%97%8F%E2%97%8F%E2%97%8F%E2%97%8F%E2%97%8F%E2%97%8F

### CODE REVIEW

**Comment by gtrufitt** [comment]
cc @guardian/dotcom-platform @guardian/design-system

File: packages/frontend/amp/components/Blocks.tsx [hunk 1]
```diff
@@ -18,7 +18,6 @@ const blockStyle = (pillar: Pillar) => css`
 
 const firstPublishedStyle = css`
     color: ${palette.neutral[7]};
-    line-height: 2rem;
```
Comment (gtrufitt) [inline packages/frontend/amp/components/Blocks.tsx]: Overridden below

File: packages/frontend/amp/components/Pagination.tsx [hunk 2]
```diff
@@ -24,7 +24,6 @@ const paginationLinkStyle = (isActive: boolean, isMarginRight: boolean) => css`
     color: ${palette.neutral[7]};
     border: 1px solid ${palette.neutral[86]};
     height: 36px;
-    line-height: 38px;
```
Comment (gtrufitt) [inline packages/frontend/amp/components/Pagination.tsx]: Had no effect, height is set and arrow is absolutely positioned

File: packages/frontend/amp/components/topMeta/TopMetaNews.tsx [hunk 3]
```diff
@@ -25,9 +25,8 @@ const headerStyle = css`
 const ratingsWrapper = css`
     background-color: ${palette.highlight.main};
     display: inline-block;
-    padding: 6px 10px 0;
```
Comment (gtrufitt) [inline packages/frontend/amp/components/topMeta/TopMetaNews.tsx]: Let padding deal with the layout, not line-height

File: packages/frontend/web/components/SubMetaLinksList.tsx [hunk 4]
```diff
@@ -77,9 +72,7 @@ export const SubMetaLinksList: React.FC<{
 }> = ({ links, isSectionLinkList, pillar }) => (
     <ul
         className={cx(subMetaLinksList, [
-            isSectionLinkList
-                ? subMetaSectionLinksList
-                : subMetaKeywordLinksList,
+            !isSectionLinkList && subMetaKeywordLinksList,
```
Comment (gtrufitt) [inline packages/frontend/web/components/SubMetaLinksList.tsx]: The biggest refactor of the lot... Basically line height was only used in the one class, so having to move from teneray to single falsey check

**Comment by SiAdcock** [comment]
Thanks so much @gtrufitt, this is really awesome work. I've added a section to the design system's information architecture to define "utilities", such as a function that can provide vertical centring.

**Comment by gtrufitt** [comment]
Thx @shtukas

**Pull request merged**